src/hotspot/share/gc/shared/generationCounters.cpp

```diff
@@ -62,29 +63,11 @@ void GenerationCounters::initialize(const char* name, int ordinal, int spaces,
   }
 }
 
-GenerationCounters::GenerationCounters(const char* name,
-                                       int ordinal, int spaces,
-                                       size_t min_capacity, size_t max_capacity,
-                                       VirtualSpace* v)
-  : _virtual_space(v) {
-  assert(v != nullptr, "don't call this constructor if v == nullptr");
-  initialize(name, ordinal, spaces,
-             min_capacity, max_capacity, v->committed_size());
-}
-
-GenerationCounters::GenerationCounters(const char* name,
-                                       int ordinal, int spaces,
-                                       size_t min_capacity, size_t max_capacity,
-                                       size_t curr_capacity)
-  : _virtual_space(nullptr) {
-  initialize(name, ordinal, spaces, min_capacity, max_capacity, curr_capacity);
-}
-
 GenerationCounters::~GenerationCounters() {
   FREE_C_HEAP_ARRAY(char, _name_space);
 }
 
-void GenerationCounters::update_all() {
-  assert(_virtual_space != nullptr, "otherwise, override this method");
-  _current_size->set_value(_virtual_space->committed_size());
+void GenerationCounters::update_all(size_t curr_capacity) {
+  _current_size->set_value(curr_capacity);
 }
```

Comment:
I don't think the `update_all` is a good method name because it only modifies the current capacity(committed size). But this method had been used since OpenJDK opened source, so we don't know its commit history before opening source. Maybe the method name `update_capacity` in the class `ZGenerationCounters` is a better name we can adopt here.

Reply:
I agree the current name can be improved to sth like `update_capacity`, but it also seems that such renaming can/should be done in its own PR, so I kept it as is.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8349077 to follow up.